src/main/java/seedu/eylah/expensesplitter/logic/commands/PaidCommand.java

```diff
@@ -12,24 +16,47 @@
 
     public static final String COMMAND_WORD = "paid";
 
-    public static final String MESSAGE_SUCCESS = "Successfully deducted amount from Person named : ";
+    public static final String MESSAGE_SUCCESS = "Successfully deducted amount from Person named: ";
 
 
-    private final Person personPaid;
+    //private final Person personPaid;
+    private final int indexOfPersonPaid;
+    private final String amountPaid;
 
     /**
      * Creates an PaidCommand to add the specified {@code Person}
      */
-    public PaidCommand (Person person) {
-        requireNonNull(person);
-        personPaid = person;
+    public PaidCommand (int indexOfPersonPaid, String amountPaid) {
+        requireNonNull(indexOfPersonPaid);
+        requireNonNull(amountPaid);
+        this.indexOfPersonPaid = indexOfPersonPaid;
+        this.amountPaid = amountPaid;
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.paidPerson(personPaid);
-        return new CommandResult(MESSAGE_SUCCESS + personPaid.getName());
+        ReadOnlyPersonAmountBook book = model.getPersonAmountBook();
+
+        //This ensures that the indexOfPersonPaid is correct.
+        if (indexOfPersonPaid < 0 || indexOfPersonPaid > book.getPersonList().size() - 1) {
+            throw new CommandException("Index of Person is incorrect. Please use `listamount` to get the index.");
+        }
+
+        Person person = book.getPersonByIndex(indexOfPersonPaid);
```

Comment:
Maybe you could name it as `personToPaid` which make the code more readable.

Reply:
Alright, you mean personToBePaid right?